### ISSUE TITLE
[Bug]: Please reconsider fix introduced in 4.16.0 for disable_api_termination 

### ISSUE THREAD

**Issue by jpbuecken** [hashicorp/terraform-provider-aws#27529]
### Terraform Core Version

1.2.4

### AWS Provider Version

4.21.0

### Affected Resource(s)

* aws_instance

### Expected Behavior

If you build your instance with disable_api_termination  enabled, it should not be possible to destroy the instance via API (even if it is terraform.

The AWS API is used in pipelines or other automatic processes / workflows. E.g. terraform is one tool to use it that way.
The idea of disable_api_protection to have an additional layer to protect your instance. 
For example it protects your instance if you have a a bug in your pipeline code or accidental/wrong usage of workflows.

If I try to destroy an instance with disable_api_termination enabled, it should fail.

### Actual Behavior

If I try to destroy an instance with disable_api_termination enabled, it succeeds.
The fix introduced in 4.16.0 enables api termination unconditionally, making the idea of option disable_api_termination meaningless.

Be aware, we use terraform modules, so we cannot use terraform prevent_destroy features as an additional layer. 
This is a long open issue in terraform (https://github.com/hashicorp/terraform/issues/18367).

So please consider if it would have been better to restore the behavior from version 3.37 as described in "Expected Behavior" in https://github.com/hashicorp/terraform-provider-aws/issues/19275

See also https://github.com/hashicorp/terraform-provider-aws/issues/19275#issuecomment-914677748

The problem people faced was that they enabled api termination via AWS console or other tools than terraform, they could not destroy the instance via terraform, because it disabled api termination again, and then tried to destroy the instance.

If I build an instance with disable_api_termination = true, then I do it on purpose to protect it for api / command line calls.
Since we use terraform only to manage our instances, we wrote workflows that change the value via terraform, then apply, and then destroy the instance, but only in places needed, to avoid accidents.

So there is no need to fix it to enable api termination unconditionally. The old behavior allows people to decide if they modify there terraform code to disable api protection (e.g. make it a variable) or use AWS console or other ways before destroy. 
For me, it looks like https://github.com/hashicorp/terraform-provider-aws/pull/19276#issuecomment-946902292 has only been declined because the code of https://github.com/hashicorp/terraform-provider-aws/pull/19277 was more up to date , but I did not read any consideration if this is a good idea.

### Terraform Configuration Files

-

### Steps to Reproduce

-	Build instance with api protection enabled
-	destroy instance
-	terraform destroy succeeded without error

**Comment by mikeplem** [comment]
If the provider functionality doesn't change would it be possible to [add a note to the documentation](https://registry.terraform.io/providers/hashicorp/aws/2.36.0/docs/resources/instance#disable_api_termination) to state that “Terraform will destroy the resource when this option is set to True.” That would give the reader more clarity about what to expect.

**Comment by bgshacklett** [comment]
Given the inflexibility of the `prevent_destroy` option in Terraform, this behavior is problematic. As has been mentioned elsewhere, if the `aws_instance` resource is within a sub-module, it makes it very difficult to manage termination protection via the `lifecycle` directive.

This certainly breaks the https://en.wikipedia.org/wiki/Principle_of_least_astonishment, and is a major change from the behavior that people have come to rely on.

**Comment by jpbuecken** [comment]
> Given the inflexibility of the prevent_destroy option in Terraform, this behavior is problematic.

First I thought "this behaviour" refers to the one I requested in this issue. But you are supporting my request, aren't you?

After I read further, "this behaviour" refers to the current behaviour introduced with 4.16.0 and the disable_api_protection option provides a protection in terraform modules.

**Comment by bgshacklett** [comment]
Yes; my apologies for the ambiguity. I've edited my previous message to clarify.

**Comment by mkyc** [comment]
I've just found that issue after I checked that this flag doesn't do what it's supposed to do. [Documentation](https://registry.terraform.io/providers/aaronfeng/aws/latest/docs/resources/instance#disable_api_termination-1) is pretty clear to me so this is bug AFAICT. 

I'd like to express my strong support for this issue (reverting back #19275). Currently setting `disable_api_termination` has no effect whatsoever. 

When working with AWS (or cloud in general) you sometimes need to make your resource hard to delete by some (too) smart automation. S3 buckets termination protection is perfect example: you really, really don't want some automation delete your bucket by mistake or even not by mistake but someone didn't give it enough thought. It saved me so many times and if I want to delete it anyway, that is complicated enough that I cannot delete something I didn't want to.

**Comment by ns-mkusper** [comment]
Similar case in my own situation: an EC2 instance with `disable_api_termination` enabled was deleted much to our surprise.

**Comment by daveadams** [comment]
Wow, just ran into this issue myself. I'm old enough to remember when the AWS provider enforced this setting on Terraform itself, so I was extra surprised when my instance with `disable_api_termination = true` was terminated without a peep. Given that the documentation does not mention that Terraform does not respect this setting for itself, I find this incredibly dangerous. Especially since similar behavior _is_ apparently enforced on other resource types.

I understand that given the different desires from the community about how this should work, that the maintainers would not want to just roll it back after three years. But at a bare minimum, the documentation should be updated to make this dangerous behavior clear.

However, there really does need to be a parameter that controls this behavior in a way that enforces its behavior on Terraform. The Terraform resource meta-parameters are insufficient to purpose since those cannot be managed via module variables.

**Comment by jpbuecken** [comment]
And the reason why developers changed the behavior is very ironic if you ask me.

If I understood the history of the change in 4.16.0 correctly, it was basically this;

- People activated API Termination Protection on the AWS web UI.
- Then people complained that terraform could not destroy the instance.

But that is exactly what the setting should do in my opinion:
If you activate API termination protection, you cannot terminate via API...

**Comment by yzargari** [comment]
Seems like [v6.8.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v6.8.0) will no longer succeed to destroy instances that have `disable_api_termination` set to `true` - unless `force_destroy` is also set to `true`
